app/javascript/components/subject/CaseStatus.js

```diff
@@ -35,32 +37,59 @@ class CaseStatus extends React.Component {
       this.setState({ [event.target.id]: value }, () => {
         if (event.target.id === 'case_status') {
           const confirmedOrProbable = value === 'Confirmed' || value === 'Probable';
-          const hideModal = this.state.isolation && confirmedOrProbable && this.props.patient.case_status !== '';
-
-          this.setState({ showCaseStatusModal: !hideModal, confirmedOrProbable }, () => {
-            // specific case where case status is just changed with no modal
-            if (hideModal) {
-              this.submit();
-            }
-            if (!confirmedOrProbable && this.state.case_status !== '') {
-              this.setState({ isolation: false });
+          this.setState({ showCaseStatusModal: true, confirmedOrProbable }, () => {
+            if (!this.props.patient.monitoring) {
+              this.setState({
+                modal_text: `Are you sure you want to change case status from ${this.props.patient.case_status} to ${value ||
+                  'blank'}? Since this record is on the Closed line list, updating this value will not move this record to another line list. If this individual should be actively monitored, please update the record’s Monitoring Status.`,
+              });
+            } else if (value === '') {
+              this.setState({
+                modal_text: `Are you sure you want to change case status from ${this.props.patient.case_status} to blank? The monitoree will remain in the same workflow.`,
+              });
+            } else if (
+              this.state.isolation &&
+              !confirmedOrProbable &&
+              (this.props.patient.case_status === 'Confirmed' || this.props.patient.case_status === 'Probable')
+            ) {
+              this.setState({
+                modal_text: `This case will be moved to the exposure workflow and will be placed in the symptomatic, non-reporting, or asymptomatic line list as appropriate to continue exposure monitoring.`,
+              });
+            } else if (confirmedOrProbable && (this.props.patient.case_status === 'Confirmed' || this.props.patient.case_status === 'Probable')) {
+              this.setState({
+                modal_text: `Are you sure you want to change the case status from ${this.props.patient.case_status} to ${value}? The record will remain in the isolation workflow.`,
+              });
+            } else if (confirmedOrProbable) {
+              this.setState({ disabled: true });
+            } else {
+              this.setState({
+                isolation: false,
+                modal_text: `The case status for the selected record will be updated to ${value} and moved to the appropriate line list in the Exposure Workflow.`,
```

Comment:
There's an additional case I found here that we should update language for. When changing from a blank Case Status to “Unknown” or “Suspect” or “Not a Case” in the _Exposure_ workflow, the modal language should say:
> The case status for the selected record will be updated to [New Status].

since the line list won't change in this case. The current language works when the record is in the isolation workflow.

Reply:
done